internal/serve/httphandler/reset_password_handler.go

```diff
@@ -32,27 +32,34 @@ func (h ResetPasswordHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	// validate request
-	v := validators.NewValidator()
-
-	v.Check(resetPasswordRequest.Password != "", "password", "password is required")
-	v.Check(resetPasswordRequest.Password != "", "reset_token", "reset token is required")
-
-	if v.HasErrors() {
-		httperror.BadRequest("request invalid", err, v.Errors).Render(w)
+	// validate password
+	badRequestExtras := map[string]interface{}{}
+	var validatePasswordError *authUtils.ValidatePasswordError
+	err = authUtils.ValidatePassword(resetPasswordRequest.Password)
+	if err != nil && errors.As(err, &validatePasswordError) {
+		for k, v := range validatePasswordError.FailedValidations() {
+			badRequestExtras[k] = v
+		}
+	}
+	// validate reset token
+	if resetPasswordRequest.ResetToken == "" {
+		badRequestExtras["reset_token"] = "reset token is required"
+	}
+	// return 400 if there are any errors
+	if len(badRequestExtras) > 0 {
+		httperror.BadRequest("request invalid", err, badRequestExtras).Render(w)
 		return
 	}
 
+	// Reset password with a valid token
 	ctx := r.Context()
-
-	// Reset password email with a valid token
 	err = h.AuthManager.ResetPassword(ctx, resetPasswordRequest.ResetToken, resetPasswordRequest.Password)
 	if err != nil {
 		if errors.Is(err, auth.ErrInvalidResetPasswordToken) {
 			httperror.BadRequest("invalid reset password token", err, nil).Render(w)
 			return
 		}
-		httperror.InternalError(ctx, "Cannot reset password", err, v.Errors).Render(w)
+		httperror.InternalError(ctx, "Cannot reset password", err, nil).Render(w)
```

Comment:
should we log the error here ?

Reply:
`InternalError` already logs it by default by calling `defaultReportErrorFunc.reportErrorFunc`:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/7f69b06ad1d0a5483bb03abae1317ee439168c59/internal/serve/httperror/httperror.go#L119

That function is configured to use the crash reporter's `LogAndReportErrors` function:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/7f69b06ad1d0a5483bb03abae1317ee439168c59/internal/serve/serve.go#L94